projects/plugins/social/src/js/components/header/index.js

```diff
@@ -36,7 +36,7 @@ const Header = () => {
 		const store = select( socialStore );
 		return {
 			connectionsAdminUrl: connectionData.adminUrl,
-			hasConnections: Object.keys( connectionData.connections || {} ).length > 0,
+			hasConnections: connectionData.length > 0,
```

Comment:
As mentioned here https://github.com/Automattic/jetpack/pull/37105/files#r1584806987

> I think we don't need any changes to this file. The existing code should work fine and also the array is in `connectionData.connections` instead of `connectionData`.